lib/service/service.go

```diff
@@ -873,10 +883,15 @@ func NewTeleport(cfg *Config, opts ...NewTeleportOption) (*TeleportProcess, erro
 			}
 			cfg.Log.Infof("Generating new host UUID: %v.", cfg.HostUUID)
 		}
-		if err := utils.WriteHostUUID(cfg.DataDir, cfg.HostUUID); err != nil {
-			if errors.Is(err, fs.ErrPermission) {
-				cfg.Log.Errorf("Teleport does not have permission to write to: %v. Ensure that you are running as a user with appropriate permissions.", cfg.DataDir)
-			}
+		// persistHostUUIDToStorages will persist the host_uuid to the local storage
+		// and to Kubernetes Secret if this process is running on a Kubernetes Cluster.
+		if err := persistHostUUIDToStorages(supervisor.ExitContext(), cfg); err != nil {
+			return nil, trace.Wrap(err)
+		}
+	} else if kubernetes.InKubeCluster() && utils.HostUUIDExistsLocaly(cfg.DataDir) {
+		// Forces the copy of the host_uuid into the Kubernetes Secret if PV storage is enabled.
+		// This is only required if PV storage is removed later.
+		if err := persistHostUUIDToStorages(supervisor.ExitContext(), cfg); err != nil {
```

Comment:
As implemented, this will try to save it to file again even though it already exists, see my comment below. Would it make sense to either update this function to not save to file when in K8s, or call writeHostUUIDToKubeSecret directly here?